### PR TITLE
[RFC] Enable -Wconversion. (1)

### PR DESCRIPTION
**Pull request by elmart**
Enable -Wconversion in 5 files.
32 files remaining.

### CODE REVIEW

**Comment by elmart** [comment]
Travis caught some warnings fired only by gcc. I'll be fixing them soon.

File: src/nvim/mark.c [hunk 1]
```diff
@@ -1499,12 +1501,13 @@ void copy_viminfo_marks(vir_T *virp, FILE *fp_out, int count, int eof, int flags
       if (load_marks) {
         if (line[1] != NUL) {
           int64_t lnum_64;
-          unsigned u;
+          unsigned int u;
```
Comment (Pyrohh) [inline src/nvim/mark.c]: This is only stylistic, but why do you say `unsigned int` as opposed to `unsigned`? I don't do it myself for brevity, but then again it makes sense for consistency with types like `unsigned long` and `unsigned char`.

Comment (elmart) [inline src/nvim/mark.c]: Well, you said it all ;-)
In new/refactored code, we try to use the complete type name.

Comment (Pyrohh) [inline src/nvim/mark.c]: Well I guess I'll start doing that then :smiley:

**Comment by elmart** [comment]
Note: First commit (ui.c) had just 1 warning when I first did it. Now, after changes in #1820, it has none. I have maintained, though, some refactoring I had done to fix the former warning. I consider that refactoring is good anyway.

File: src/nvim/mbyte.c [hunk 2]
```diff
@@ -1375,7 +1375,7 @@ static int dbcs_ptr2char(const char_u *p)
  */
 int utf_ptr2char(const char_u *p)
 {
-  int len;
+  uint8_t len;
```
Comment (Pyrohh) [inline src/nvim/mbyte.c]: Since you're already doing it in other places, this could be moved down to the initialization.

**Comment by justinmk** [comment]
merged. Thanks!

**Pull request closed without merging**